src/components/Homepage/HomePage.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { Link } from "react-router-dom";
+const axios = require("axios");
+
+const api_key = process.env.REACT_APP_API_KEY;
+
+export default function HomePage({ setMovieInfo }) {
+  const [movies, setMovies] = useState([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchMovieData(id) {
+    const response = await axios.get(
+      `https://api.themoviedb.org/3/movie/${id}`,
+      {
+        params: {
+          api_key: api_key,
+        },
+      }
+    );
+    console.log(response);
+    setMovieInfo(response.data);
+  }
+
+  async function fetchData() {
+    const response = await axios.get(
+      "https://api.themoviedb.org/3/movie/popular",
+      {
+        params: {
+          api_key: api_key,
+        },
+      }
+    );
+    setMovies(response.data.results);
+  }
+
+  return (
+    <>
+      <h1 className="text-4xl mt-10">Most popular movies</h1>
+      <div className="mt-20 grid gap-y-10 grid-cols-5">
+        {movies.map((movie) => (
```

Comment:
You should add loading state. When the data is loading, the page should NOT be empty. Please add spinner.

Reply:
fixed